goblas/level2double.go

```diff
@@ -1683,6 +1683,10 @@ func (Blas) Dsyr2(ul blas.Uplo, n int, alpha float64, x []float64, incX int, y [
 	if incY == 0 {
 		panic(zeroInc)
 	}
+	if alpha == 0 {
```

Comment:
No, those panics are informative; the state of alpha should not change whether we expect data slices be be appropriately long for the given parameters.